src/fastgltf.cpp

```diff
@@ -5502,6 +5960,95 @@ void fg::Exporter::writeNodes(const Asset& asset, std::string& json) {
 				if (json.back() != '{') json += ',';
 				json += R"("KHR_lights_punctual":{"light":)" + std::to_string(it->lightIndex.value()) + "}";
 			}
+
+#if FASTGLTF_ENABLE_KHR_PHYSICS_RIGID_BODIES
+			if (it->physicsRigidBody.has_value()) {
+				if (json.back() != '{') json += ',';
+				json += R"("KHR_physics_rigid_body":{)";
+				if (it->physicsRigidBody->motion.has_value()) {
+					const auto& motion = *it->physicsRigidBody->motion;
+					json += R"("motion":{"isKinematic":)" + std::to_string(motion.isKinematic);
+					if (motion.mass.has_value()) {
+						json += R"(,"mass":)" + std::to_string(*motion.mass);
+					}
+					json += R"(,"centerOfMass":[)" + to_string(motion.centerOfMass) + "]";
+					if (motion.inertialDiagonal.has_value()) {
+						json += R"(,"inertialDiagonal":[)" + to_string(*motion.inertialDiagonal) + "]";
+					}
+					if (motion.inertialOrientation.has_value()) {
+						json += R"(,"inertialOrientation":[)" + to_string(*motion.inertialOrientation) + "]";
+					}
+					json += R"(,"linearVelocity":)" + to_string(motion.linearVelocity)
+				        + R"(],"angularVelocity":[)" + to_string(motion.angularVelocity)
+				        + R"(],"gravityFactor":[)" + to_string_fp(motion.gravityFactor) + "}";
+				}
+
+				if (it->physicsRigidBody->trigger.has_value()) {
+					if (json.back() != '{') json += ',';
+					const auto& trigger = *it->physicsRigidBody->trigger;
+					json += R"("trigger":{)";
+					std::visit(visitor{
+	            [&](auto&) {
+							// Is this needed?
```

Comment:
Nope, and this should probably use `fastgltf::visit_exhaustive`, which checks if you have all the overloads specified correctly, and aren't missing any, have wrong const-ness, or ... by accident.